inst/rmd/scratch_gimap_lfc_review.Rmd

```diff
@@ -0,0 +1,99 @@
+---
+title: "scratch gimap LFC & Normalization Review"
+output: html_document
+date: "`r Sys.Date()`"
+---
+
+```{r warning=FALSE, message=FALSE}
+library(tidyverse)
+library(devtools)
+library(magrittr)
+library(ggpubr)
+
+devtools::load_all()
+```
+
+## Get the gimap Normalization/LFC results
+
+```{r}
+ gimap_dataset <- get_example_data("gimap") %>%
+   gimap_filter() %>%
+   gimap_annotate() %>%
+   gimap_normalize(
+     timepoints = "day",
+     replicates = "rep")
+```
+
+## Load the GI Mapping Normalization/LFC results
+
+```{r}
+old_lfc_results <- read_tsv("~/Downloads/results/calculate_LFC/tables/tsv/HeLa_lfc_annot_adj_pgRNA.txt")
+```
+
+## Compare them
+
+### Rows
+
+```{r}
+nrow(gimap_dataset$normalized_log_fc)
+nrow(old_lfc_results)
+setdiff(old_lfc_results$pgRNA_id, gimap_dataset$normalized_log_fc$pg_ids)
+sum(gimap_dataset$normalized_log_fc$pg_ids == old_lfc_results$pgRNA_id)
+```
+
+The number of rows is more than 31817 which is the number of unique pgRNA IDs previously in the pipeline., but there aren't any pgRNA IDs present in one and not the other and the IDs seem to match row-wise. Each pgRNA ID is there in triplicate maybe? -- yes that seems to be the case because `unique(table(gimap_dataset$normalized_log_fc$pg_ids))` gives: `r unique(table(gimap_dataset$normalized_log_fc$pg_ids))`. Oh that's because there are 3 reps for each (A, B, and C for `old_lfc_results` and late_RepA, late_RepB, and late_RepC for `gimap_dataset$normalized_log_fc`) 
+
+#### sync the reps values
+
+```{r}
+old_lfc_results %<>% mutate(
+  rep = recode(rep, "A" = "late_RepA", "B" = "late_RepB", "C" = "late_RepC")
+)
+```
+
+### Columns
+
+```{r}
+ncol(gimap_dataset$normalized_log_fc)
+ncol(old_lfc_results)
+print("Column names in new results not in old results")
+setdiff(colnames(gimap_dataset$normalized_log_fc), colnames(old_lfc_results))
+print("Column names in old results not in new results")
+setdiff(colnames(old_lfc_results), colnames(gimap_dataset$normalized_log_fc))
+```
+| New results column name | Equivalent in old results |
+|:-----------------------:|:-------------------------:|
+| `pg_ids`                | `pgRNA_id`                |
+| `log2_cn_gene1`         | `gene1_log2_cn`           |
+| `log2_cn_gene2`         | `gene2_log2_cn`           |
+| `log2_tpm_gene1`        | `gene1_log2_tpm`          |
+
+Are `lfc_adj` and `CRISPR_score` equivalent?
+
+`plasmid_RepA`, `early_RepA`, `lfc_plasmid_vs_late` `unexpressed_control_flag` seem to be columns unique to the new results, not in the old results
+
+`gene1_entrez_id` and `gene2_entrez_id` seem to be columns unique to the old results, not in the new results. (NOTE: but there are ensembl IDs in both, so I don't think this is a concern.)
+
+```{r}
+joindf <- dplyr::full_join(old_lfc_results, gimap_dataset$normalized_log_fc,
+                           by = c("pgRNA_id" = "pg_ids", "rep"),
+                           suffix = c("_old", "_new")) %>%
+  select(lfc_adj, CRISPR_score, rep) %>%
```

Comment:
Just to be clear lfc_adj is the column in the old data that is actually the CRISPR score? So we are comparing CRISPR scores to CRISPR scores?

Reply:
No, this is the part I'm not sure about. Looking at the GI_Mapping code, [the `CRISPR_score` column is a rename of the `lfc_adj3` column]( https://github.com/FredHutch/GI_mapping/blob/41ac7d5ed7025252343e2c823fba22f8a363e25c/workflow/scripts/03-filter_and_calculate_LFC.Rmd#L620C3-L620C34). I use the CRISPR_score column from the GI_Mapping results and compare that to the `lfc_adj` column from the `gimap` results. If `CRISPR_score` from GI_Mapping doesn't correspond to `lfc_adj` from gimap, which one does. I did some code comparing this afternoon and I think it'll be [the `lfc_adj2` column from GI_Mapping](https://github.com/FredHutch/GI_mapping/blob/41ac7d5ed7025252343e2c823fba22f8a363e25c/workflow/scripts/03-filter_and_calculate_LFC.Rmd#L451) as it has a similar calculation as [`lfc_adj` in gimap](https://github.com/FredHutch/gimap/blob/04413b51ff483946b47e6e8c30f082b6d6c7655d/R/04-normalize.R#L140C7-L140C138)

Reply:
Ah in this case from our data what you want is the CRISPR score which you can get from `gimap_dataset$crispr_score` if `calc_crispr()` has been ran. 

That should be a more apples to apples comparison. 

It would also still be worth comparing precursor lfc's to our lfc's as well but perhaps starting with CRISPR is good start.